### PR TITLE
make lint: extend checks

### PR DESCRIPTION
**Pull request by vrothberg**
This PR includes a series of commits to extend the linter checks to include:
* `pkg/tracing`
* `docs/`
* unit tests (i.e., `_test.go` files)

Especially checking the unit tests could make CI more robust as rigorous error checks could reveal flakes.

### CODE REVIEW

**Comment by vrothberg** [comment]
Next round.  @TomSweeneyRedHat @rhatdan @baude @mheon PTAL

The next candidate (if time allows) is to include `test/` into the checks as well. After that, we can enable the currently disabled one-by-one.

File: cmd/podman/remoteclientconfig/configfile_test.go [hunk 1]
```diff
@@ -92,14 +92,15 @@ func TestReadRemoteConfig(t *testing.T) {
 		{"nouser", args{reader: strings.NewReader(noUser)}, makeNoUserResult(), false},
 	}
 	for _, tt := range tests {
+		test := tt
```
Comment (TomSweeneyRedHat) [inline cmd/podman/remoteclientconfig/configfile_test.go]: why keep 'tt' at all, can't you drop this line and replace tt in line 94 with 'test'?

File: cmd/podman/remoteclientconfig/configfile_test.go [hunk 2]
```diff
@@ -192,17 +194,18 @@ func TestRemoteConfig_GetRemoteConnection(t *testing.T) {
 		{"none", fields{Connections: blank}, args{name: "foobar"}, nil, true},
 	}
 	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
+		test := tt
```
Comment (TomSweeneyRedHat) [inline cmd/podman/remoteclientconfig/configfile_test.go]: ditto tt comment

File: libpod/image/image_test.go [hunk 3]
```diff
@@ -213,7 +210,7 @@ func TestImage_RepoDigests(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	for _, test := range []struct {
+	for _, tt := range []struct {
```
Comment (TomSweeneyRedHat) [inline libpod/image/image_test.go]: OK, now I'm confused.  Lint wanted tt changed to test in the other code, but test to tt here?

File: cmd/podman/remoteclientconfig/configfile_test.go (hunk 1 above)
Comment (vrothberg) [reply cmd/podman/remoteclientconfig/configfile_test.go]: We shouldn't use loop variables in go which may leak outside the loop (e.g., by appending it to a slice outside of the loop) or when the code can run in parallel. The latter is the case for all t.Run(..)s as unit tests could run in parallel.

Comment (vrothberg) [reply cmd/podman/remoteclientconfig/configfile_test.go]: The "problem" with loop variables is that they're actually a pointer. `test := tt` allocates a new variable and copies the value of `tt`. This way, we're safe.

File: libpod/image/image_test.go (hunk 3 above)
Comment (vrothberg) [reply libpod/image/image_test.go]: I wanted it to be consistent. `tt` as the loop variable, `test` for the variable we're using in the loop's body.

File: pkg/hooks/1.0.0/when_test.go [hunk 4]
```diff
@@ -48,7 +52,8 @@ func TestHasBindMountsAnd(t *testing.T) {
 	hasBindMounts := true
 	when := When{HasBindMounts: &hasBindMounts}
 	config := &rspec.Spec{}
-	for _, containerHasBindMounts := range []bool{false, true} {
+	for _, b := range []bool{false, true} {
+		containerHasBindMounts := b
```
Comment (TomSweeneyRedHat) [inline pkg/hooks/1.0.0/when_test.go]: OK, these three b, a, and o are like tt/tests above.  I don't get it.

File: cmd/podman/remoteclientconfig/configfile_test.go (hunk 1 above)
Comment (TomSweeneyRedHat) [reply cmd/podman/remoteclientconfig/configfile_test.go]: thx, didn't see the leak outside of the loop happening in this code, but may well have missed it.

**Comment by rhatdan** [comment]
LGTM
Nice work.

File: .gitignore [hunk 5]
```diff
@@ -21,11 +21,11 @@ __pycache__
 /cmd/podman/varlink/ioprojectatomicpodman.go
 /cmd/podman/varlink/iopodman.go
 .gopathok
-test/e2e/e2e.coverprofile
```
Comment (mheon) [inline .gitignore]: Why? I think this is useful to prevent it accidentally being committed

Comment (vrothberg) [reply .gitignore]: Because we ignore all `*.coverprofile` now

**Comment by mheon** [comment]
The only thing pending is a Prow test, so I'll go ahead and whack this
/lgtm

**Comment by haircommander** [comment]
git hooks have been dropping for prow btw, the suggestion is to spam retest until it triggers
/retest

**Pull request merged**